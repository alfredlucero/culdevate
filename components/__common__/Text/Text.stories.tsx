import React from "react";
import { storiesOf } from "@storybook/react";
import StoryFrame from "../StoryFrame";
import { H1, H2, H3, P } from "./index";

storiesOf("Text", module).add("H1, H2, H3, P", () => (
  <StoryFrame>
    <H1>Culdevate Header 1 Text</H1>
    <H2>Culdevate Header 2 Text</H2>
    <H3>Culdevate Header 3 Text</H3>
    <P>Culdevate Paragraph Text</P>
    <P small>Culdevate Paragraph Small Text</P>
  </StoryFrame>
));
