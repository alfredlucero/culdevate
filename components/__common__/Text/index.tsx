import styled from "styled-components";
import { culdevateDefaultTheme } from "../../defaultTheme";

const H1 = styled.h1`
  margin: 0;
  padding: 0;
  font-family: ${props => props.theme.fonts.headerFontFamily};
  font-size: 3rem;
  line-height: 2;
  color: ${props => props.theme.colors.teal};
`;

H1.defaultProps = {
  theme: culdevateDefaultTheme,
};

const H2 = styled.h2`
  margin: 0;
  padding: 0;
  font-family: ${props => props.theme.fonts.headerFontFamily};
  font-size: 2.5rem;
  line-height: 2;
  color: ${props => props.theme.colors.teal};
`;

H2.defaultProps = {
  theme: culdevateDefaultTheme,
};

const H3 = styled.h3`
  margin: 0;
  padding: 0;
  font-family: ${props => props.theme.fonts.headerFontFamily};
  font-size: 2rem;
  line-height: 2;
  color: ${props => props.theme.colors.teal};
`;

H3.defaultProps = {
  theme: culdevateDefaultTheme,
};

const H4 = styled.h4`
  margin: 0;
  padding: 0;
  font-family: ${props => props.theme.fonts.headerFontFamily};
  font-size: 1.75rem;
  line-height: 2;
  color: ${props => props.theme.colors.orange};
`;

H4.defaultProps = {
  theme: culdevateDefaultTheme,
};

interface PProps {
  small?: boolean;
}

const P = styled.p<PProps>`
  margin: 0;
  padding: 0;
  font-family: ${props => props.theme.fonts.bodyFontFamily};
  font-size: ${props => (props.small ? "1.25rem" : "1.5rem")};
  line-height: 1.5;
  color: ${props => props.theme.colors.gray};
`;

P.defaultProps = {
  theme: culdevateDefaultTheme,
};

export { P, H1, H2, H3, H4 };
