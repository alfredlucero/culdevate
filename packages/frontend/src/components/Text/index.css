.text {
  @apply font-sans text-gray-800;
}

.text-normal {
  @apply text-base;
}

.text-small {
  @apply text-sm;
}

.text-bold {
  @apply font-bold;
}

.text-italic {
  @apply italic;
}
