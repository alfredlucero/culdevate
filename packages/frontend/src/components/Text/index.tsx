import React from "react";
import cn from "classnames";
import { CommonProps } from "../commonProps";

export interface TextProps extends CommonProps {
  variant: TextVariant;
  children: React.ReactNode;
  small?: boolean;
  italic?: boolean;
  bold?: boolean;
}

type TextVariant = "p" | "span";

const textBaseClasses = ["font-sans", "text-gray-800", "text-base"];

const Text: React.FC<TextProps> = ({
  variant,
  children,
  small = false,
  italic = false,
  bold = false,
  testId = "",
  className = "",
  ...passThroughProps
}) => {
  if (variant === "span") {
    return (
      <span
        className={`${cn(textBaseClasses, {
          "text-sm": small,
          italic: italic,
          "font-bold": bold,
        })} ${className}`}
        {...(testId !== "" ? { "data-testid": testId } : {})}
        {...passThroughProps}
      >
        {children}
      </span>
    );
  }

  if (variant === "p") {
    return (
      <p
        className={`${cn(textBaseClasses, {
          "text-sm": small,
          italic: italic,
          "font-bold": bold,
        })} ${className}`}
        {...(testId !== "" ? { "data-testid": testId } : {})}
        {...passThroughProps}
      >
        {children}
      </p>
    );
  }

  return null;
};

export default Text;
