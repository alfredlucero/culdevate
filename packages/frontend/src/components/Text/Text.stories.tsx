import React from "react";
import { storiesOf } from "@storybook/react";
import Text from "./index";

storiesOf("Common/Text", module)
  .add("Paragraph Variant", () => <Text variant="p">Paragraph Variant</Text>)
  .add("Paragraph Bold", () => (
    <Text variant="p" bold={true}>
      Paragraph Bold
    </Text>
  ))
  .add("Paragraph Italic", () => (
    <Text variant="p" italic={true}>
      Paragraph Italic
    </Text>
  ))
  .add("Paragraph Small", () => (
    <Text variant="p" small={true}>
      Paragraph Small
    </Text>
  ))
  .add("Span Variant", () => (
    <Text variant="span" small={true}>
      Span Variant
    </Text>
  ));
