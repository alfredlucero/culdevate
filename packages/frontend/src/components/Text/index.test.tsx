import React from "react";
import { render } from "@testing-library/react";
import Text from "./index";

describe("<Text />", () => {
  const textTestId = "textTestId";

  test("should render paragraph variant without error", () => {
    const { container } = render(
      <Text variant="p" testId={textTestId} className="extra-text-class">
        Paragraph Content
      </Text>,
    );

    expect(container).toMatchSnapshot();
  });

  test("should render span variant without error", () => {
    const { container } = render(
      <Text variant="span" testId={textTestId} className="extra-text-class">
        Span Content
      </Text>,
    );

    expect(container).toMatchSnapshot();
  });

  test("should render small class when small", () => {
    const { getByTestId } = render(
      <Text variant="p" small={true} testId={textTestId}>
        Paragraph Content
      </Text>,
    );

    expect(getByTestId(textTestId)).toHaveClass("text-small");
  });

  test("should render bold class when bold", () => {
    const { getByTestId } = render(
      <Text variant="p" bold={true} testId={textTestId}>
        Paragraph Content
      </Text>,
    );

    expect(getByTestId(textTestId)).toHaveClass("text-bold");
  });

  test("should render italic class when italic", () => {
    const { getByTestId } = render(
      <Text variant="p" italic={true} testId={textTestId}>
        Paragraph Content
      </Text>,
    );

    expect(getByTestId(textTestId)).toHaveClass("text-italic");
  });
});
