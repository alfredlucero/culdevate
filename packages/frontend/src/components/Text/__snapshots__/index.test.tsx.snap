// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render paragraph variant without error 1`] = `
<div>
  <p
    class="text text-normal extra-text-class"
    data-testid="textTestId"
  >
    Paragraph Content
  </p>
</div>
`;

exports[`<Text /> should render span variant without error 1`] = `
<div>
  <span
    class="text text-normal extra-text-class"
    data-testid="textTestId"
  >
    Span Content
  </span>
</div>
`;
